ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -0,0 +1,255 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Shapes 1.15
+
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+import shared.stores.send 1.0
+import shared.popups.send.views 1.0
+import shared.popups.send.panels 1.0
+
+import utils 1.0
+
+Control {
+    id: root
+
+    // input API
+    required property TransactionStore store // TODO refactor to accept SwapModalAdaptor/SwapStore ???
+
+    property string inputTokenKey
+    onInputTokenKeyChanged: Qt.callLater(d.setSelectedHoldingId, inputTokenKey, Constants.TokenType.ERC20)
+    property string inputTokenAmount
+    onInputTokenAmountChanged: Qt.callLater(() => amountToSendInput.input.text =
+                                            Number(inputTokenAmount).toLocaleString(Qt.locale(), 'f', -128))
+
+    property int swapSide: SwapInputPanel.SwapSide.Pay
+
+    // output API
+    readonly property string selectedHoldingId: d.selectedHoldingId
```

Comment:
this is from holding selector and I really thought we'd also rework that part because the current implementation doesn't feel very clean. not sure what you think? I think it should be done , cause its again going to be used in send and swap.

Reply:
Yeah would be definitely nice but; avoid it doing it here though but rather as a separate task, as that is going to involve adjusting the SendModal too which might be too disruptive and out of scope of this task